build/assets/tuned/openshift-node-performance

```diff
@@ -18,6 +18,9 @@ isolated_cores={{.IsolatedCpus}}
 {{end}}
 
 not_isolated_cores_expanded=${f:cpulist_invert:${isolated_cores_expanded}}
+# minimum kernel version applicable for enabling stalld is 4.18.0-240.22.1.el8_3.x86_64 or 4.18.0-240.22.1.rt7.77.el8_3.x86_64
+# starting kernel version for oc 4.7 is 4.18.0-240.10.1.el8_3.x86_64
+stalld_minimum_kernel_version_prefix=^Linux [^ ]+ (?!4\.18\.0-240\.([0-9]|1[0-9]|2[0-1])\.)
```

Comment:
I wonder if there is a way to test this regex besides the minimal coverage the e2e tests give us

Reply:
Yes, but it should be under tuned in my opinion

Reply:
Surely the feature itself needs to be tested in tuned, but the fact that this regex captures properly the range of versions we care about seem to belong here more than in tuned.

Reply:
We should definitely have a test in PAO that checks what we enable / not enable based on kernel version

Reply:
We already have one for starting stalld, maybe we can add another one for not-starting it.. I just wonder how to inject different uname

Reply:
we might be able to install an older kernel version , then reinstall the current one but that could really tamper the test environment and take a lot of time.  
another option might be adding a test suit for overriding tuned yamls as we do for hotfixes and add a different regex that will cause stalld to be disabled but that won't be a direct test to what we have under the default tuned base profile.

Reply:
Can we trick tuned to think the uname is different? By replacing the uname binary or something?

Reply:
AFAIU tuned just calls `os.uname()` which is a cpython builtin. Maybe worth checking with tuned folks about the cleanest way to run this check.

Reply:
*MAYBE* something like https://github.com/fromanirh/fromanirh/tree/main/tools/fake-uname ?

Reply:
Yeah, I was thinking about ld preload as well, but all our tests are part of the same binary, right? So we would need a way to change it for separate tests without restarting the binary.

Reply:
Well a preload shlib can be made smarter - it's a testing tool after all, it will never see production - like reading a file each time, but we should get in touch with tuned folks and see if a code change on their side can make the testing easier. OTOH we'll have a tech debt until we can really test this flow.

Reply:
Or maybe we can just (ab)use the tuned `uname_string` configuration option:
https://github.com/redhat-performance/tuned/blob/master/tuned-main.conf#L46